idaes/generic_models/properties/core/eos/ceos.py

```diff
@@ -344,6 +459,22 @@ def build_parameters(b):
                 initialize=kappa_data,
                 doc=cname+' binary interaction parameters',
                 units=None))
+        
+        if b._cubic_type == CubicType.PR:
+            func_fw = func_fw_PR
+        elif b._cubic_type == CubicType.SRK:
+            func_fw = func_fw_SRK
+        else:
+            raise BurntToast(
+                    "{} received unrecognized cubic type. This should "
+                    "never happen, so please contact the IDAES developers "
+                    "with this bug.".format(b.name))
+        setattr(param_block, cname+"_func_fw", func_fw)
```

Comment:
As above, do we need to attach most of these to the Block? If it is matter of saving repeated code, then we could write a short utility unction that can retrieve these from the config block that we call when needed.